src/mobility/components/OperatorActionButton.tsx

```diff
@@ -111,17 +112,39 @@ const OperatorActionButtonWithValueCode = ({
   buttonOnPress,
   buttonText,
 }: OperatorActionButtonWithValueCodeProps) => {
-  const {data: valueCode, isLoading: isLoadingValueCode} =
-    useValueCodeQuery(operatorId);
+  const {t} = useTranslation();
 
-  if (isLoadingValueCode) {
+  const {
+    mutate: fetchValueCode,
+    isLoading: isFetchingValueCode,
+    isError: isFetchingValueCodeError,
+  } = useValueCodeMutation(operatorId);
+
+  const appSwitchButtonOnPress = useCallback(() => {
+    fetchValueCode(undefined, {
+      onSuccess: (valueCode) => valueCode && buttonOnPress(valueCode),
```

Comment:
Not sure if using `onSuccess` is better than calling `mutateAsync()` and use `then`. 🤔 

My thought is, since a `Promise` with `then` is built-in JS, it is less prone to error or API changes, compared to using `onSuccess` call through the `useMutation()` API. 

What do you think?

Reply:
Good thinking. Could be more readable as well I guess :+1:

Reply:
https://github.com/AtB-AS/mittatb-app/pull/4569/commits/93bccf37542eaa2fcbfcd1f03fde46a72e46a289